packages/starknet-snap/src/rpcs/declare-contract.ts

```diff
@@ -0,0 +1,223 @@
+import type { Component } from '@metamask/snaps-sdk';
+import { row, text } from '@metamask/snaps-sdk';
+import convert from 'ethereum-unit-converter';
+import type { Infer } from 'superstruct';
+import { assign, object, optional, string } from 'superstruct';
+
+import {
+  AddressStruct,
+  BaseRequestStruct,
+  DeclareContractPayloadStruct,
+  mapDeprecatedParams,
+  UniversalDetailsStruct,
+  confirmDialog,
+  AccountRpcController,
+} from '../utils';
+import { UserRejectedOpError } from '../utils/exceptions';
+import { declareContract as declareContractUtil } from '../utils/starknetUtils';
+
+// Define the DeclareContractRequestStruct
+export const DeclareContractRequestStruct = assign(
+  object({
+    address: AddressStruct, // Sender address
+    contractPayload: DeclareContractPayloadStruct, // Contract payload structure
+    details: optional(UniversalDetailsStruct), // Optional invocation details
+  }),
+  BaseRequestStruct, // Base request struct, could include chainId, etc.
+);
+
+export const DeclareContractResponseStruct = object({
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  transaction_hash: string(),
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  class_hash: string(),
+});
+
+export type DeclareContractParams = Infer<typeof DeclareContractRequestStruct>;
+export type DeclareContractResponse = Infer<
+  typeof DeclareContractResponseStruct
+>;
+
+/**
+ * The RPC handler to declare a contract.
+ */
+export class DeclareContractRpc extends AccountRpcController<
+  DeclareContractParams,
+  DeclareContractResponse
+> {
+  protected requestStruct = DeclareContractRequestStruct;
+
+  protected responseStruct = DeclareContractResponseStruct;
+
+  protected async preExecute(params: DeclareContractParams): Promise<void> {
+    // Define mappings to ensure backward compatibility with previous versions of the API.
+    // These mappings replace deprecated parameter names with the updated equivalents,
+    // allowing older integrations to function without changes
+    const paramMappings: Record<string, string> = {
+      senderAddress: 'address',
+      invocationsDetails: 'details',
+    };
```

Comment:
nit:

if we accept above change, that also required to update too
```suggestion
    const paramMappings: Record<string, string> = {
      senderAddress: 'address',
      invocationsDetails: 'details',
      contractPayload: 'payload'
    };
```